contracts/data/WitnetBoardData.sol

```diff
@@ -2,9 +2,55 @@
 
 pragma solidity >=0.7.0 <0.9.0;
 
+import "../exports/WitnetTypes.sol";
 
 /**
  * @title Witnet Board basal data model. 
  * @author Witnet Foundation
  */
-abstract contract WitnetBoardData {}
+abstract contract WitnetBoardData {  
+
+  struct SWitnetBoardData {
+    address owner;
+    address instance;
+    mapping (uint => SWitnetBoardDataRequest) requests;
+    uint256 noRequests;
+  }
+
+  struct SWitnetBoardDataRequest {
+    WitnetTypes.DataRequest dr;
+    bytes result;
+  }
+
+  constructor() {
+    __data().owner = msg.sender;
+  }
+
+  modifier onlyOwner {
+    require(msg.sender == __data().owner, "WitnetBoardData: only owner");
+    _;    
+  }
+
+  modifier resultNotYetReported(uint256 id) {
+    require(__dataRequest(id).txhash == 0, "WitnetBoardData: already solved");
+    _;
+  }
+
+  modifier wasPosted(uint256 id) {
+    require(id > 0 && id <= __data().noRequests, "WitnetBoardData: not yet posted");
+    _;
+  }
+
+  function __data() internal pure returns (SWitnetBoardData storage _struct) {
+    assembly {
+      _struct.slot := WITNET_BOARD_DATA_SLOTHASH
+    }
+  }
+
+  function __dataRequest(uint256 id) internal view returns (WitnetTypes.DataRequest storage) {
+    return __data().requests[id].dr;
+  }
+  
+  bytes32 internal constant WITNET_BOARD_DATA_SLOTHASH = "WitnetBoard.Data";  
```

Comment:
I don't understand what is this, it's called "hash" but it is a plaintext string?

Reply:
The text "WitnetBoard.Data" is actually converted to `bytes32`, which aftewards is interpreted as a slot storage pointer (ie. storage slot hash). Therefore, maximum name length would be 32.

This approach enables the possibility to let a contract report supported storage slots in a human-readable way, without the caller needing a lookup service (either onchain or offchain) as to translate hashes to actual data struct names. 

Two things to have in mind, though: 
1. Storage slot name maximum length is 32.
2. Actual storage slot size must be smaller than `2 ^ (32 - len(slot_name) - 1` if a storage slot name is to be eventually reused in further data model extensions (e.g. `WitnetBoard.Data2`).